### PR TITLE
fix: Update OpenFeature Provider user ID mapping priority and add userId support

### PR DESCRIPTION
**Pull request by jonathannorris**
### Summary
Enhances the OpenFeature Provider to support `userId` field mapping and corrects the priority order to match other DevCycle SDKs.

### Changes
- **Priority Order**: Updated to `targetingKey` → `user_id` → `userId` (was previously backwards)
- **New Field Support**: Added support for `userId` field mapping from OpenFeature evaluation context
- **Validation**: Added robust validation for user ID fields (required, must be string, non-empty)
- **Error Handling**: Improved error messages to include specific field names for better debugging
- **Custom Data**: All user ID fields (`targetingKey`, `user_id`, `userId`) are now excluded from custom data

### Validation Added
- Throws `ArgumentError` if no user ID field is provided
- Throws `ArgumentError` if user ID field is not a string (with field name)
- Throws `ArgumentError` if user ID field is empty

### Tests
- ✅ 45 comprehensive tests covering all scenarios
- ✅ Priority order validation
- ✅ Field validation edge cases  
- ✅ Custom data exclusion behavior
- ✅ Backward compatibility

This change aligns the Ruby SDK with other DevCycle OpenFeature providers and ensures consistent behavior across all platforms.

---

**Open Background Agent:** 
[Web](https://www.cursor.com/agents%3Fid=bc-7df500b8-264b-4ab7-82ad-bd4309776a0f) · [Cursor](https://cursor.com/background-agent%3FbcId=bc-7df500b8-264b-4ab7-82ad-bd4309776a0f)

[Slack Thread](https://taplytics.slack.com/archives/C02DU5GTB6U/p1752872862815949%3Fthread_ts=1752872862.815949

### CODE REVIEW

File: lib/devcycle-ruby-server-sdk/api/dev_cycle_provider.rb [hunk 1]
```diff
@@ -48,15 +48,42 @@ def self.user_from_openfeature_context(context)
...
+      end
+      
+      unless user_id.is_a?(String)
+        raise ArgumentError, "User ID field '#{user_id_field}' must be a string, got #{user_id.class}"
+      end
+      
+      # Check after type validation to avoid NoMethodError on non-strings
+      if user_id.empty?
+        raise ArgumentError, "User ID is required. Must provide one of: targeting_key, user_id, or userId"
+      end
```
[the remark this answers is not in the document]
Comment (jonathannorris) [reply lib/devcycle-ruby-server-sdk/api/dev_cycle_provider.rb:78]: I'm confused, you were complaing they all weren't verbose... so I made them all verbose

[the remark this answers is not in the document]
Comment (JamieSinn) [reply lib/devcycle-ruby-server-sdk/api/dev_cycle_provider.rb:78]: I was reviewing an older copy. My mistake

**Review by JamieSinn — APPROVED** [review]

**Review by suthar26 — APPROVED** [review]
